python/pyiceberg/expressions/visitors.py

```diff
@@ -517,7 +519,7 @@ def visit_equal(self, term: BoundTerm, literal: Literal[Any]) -> bool:
         pos = term.ref().accessor.position
         field = self.partition_fields[pos]
 
-        if field.lower_bound is None:
+        if field.lower_bound is None or field.upper_bound is None:
```

Comment:
Great catch! Thanks!

Reply:
If one is `None` then both should be, so this shouldn't have any effect. But it's fine to update this.

Reply:
This is required for the linter since we're the variable below.